content/architecture/documentation/index.md

```diff
@@ -0,0 +1,267 @@
+* Passed into this command will be the path to the CSV file containing endpoints to be polled for this collection.
+* For each poll, if the poll doesn't terminate in an error condition, the raw payloads fetched will be saved in `collection/log/<iso-date>/<hash>.<extension>`
+  * where `<iso-date>` is the [ISO 8601 Calendar date with day of month](https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates) and `<hash>` is the `SHA256` hash of the content returned from the poll.
+
+#### Commit collection
+
+* The `collection` folder, containing the raw payloads fetched in the previous step are committed to the collection repository, and pushed to the upstream remote in Github
+
+#### Update the collection dataset
+
+* This step updates the `collection/log.csv` and `collection/resource.csv` files
+  * `collection/resource.csv` contains a record of the latest attempt to update a collection, with the hash and size of the result, along with organization type, endpoints polled and start and end date
+  * `collection/log.csv` contains records corresponding to every attempt made to poll and endpoint, the content type, status code, start date, end date and time taken
+* A new make entrypoint is added to the local `collection/pipeline.mk` which serves this dataset
+
+#### Commit updated collection
+
+* The `collection` folder, containing the files modified in the previous step are committed to the collection repository, and pushed to the upstream remote in Github
+
+#### Push collection files to S3
+
+* The `collection` folder, containing the files modified in the previous steps are pushed up to the collection S3 bucket
+
+#### Run the pipeline to make the dataset
+
+* The `make dataset` entrypoint calls through to the `build-dataset` and `run-pipeline` `make` macros.
+<!-- TODO figure how how it actually does this  -->
+* Three [`digital-land` CLI](https://github.com/digital-land/digital-land-python) entrypoints seem to be called off the back of this
+  * `pipeline` orchestrates the transformation pipeline that runs on the data
+    <!-- * It will instantiate pipeline-specific instances of: -->
+      <!-- * `Organisation` -->
+      <!-- * `Issues` -->
+      <!-- * `Collection` -->
+      <!-- * `patches` -->
+      <!-- * `Converter` -->
+      <!-- * `Normaliser` -->
+      <!-- * `Mapper` -->
+      <!-- * `Filterer` -->
+      <!-- * `Harmoniser` -->
+      <!-- * `Transformer` -->
+      <!-- * `Lookup` -->
+      <!-- * `Slugger` -->
+    * It will operate on the data using the following (in-order) operations:
+      * [`convert`](https://digital-land.github.io/guidance/pipeline/#2-convert) - Converts all data to CSV format and exposes as a stream
+      * [`normalise`](https://digital-land.github.io/guidance/pipeline/#3-normalise) - Normalise whitespace, strip null characters & blank lines
+      * `line_conversion` - ?
```

Comment:
Toby to query Paul D about what a good generalised explanation for the purpose of this stage is